### PR TITLE
*: Add an option to stop writting binlog when meets binlog error

### PR DESCRIPTION
**Pull request by tiancaiamao**
PTAL @coocood @shenli

### CODE REVIEW

**Comment by coocood** [comment]
@tiancaiamao 
Please add tests

File: sessionctx/binloginfo/binloginfo.go [hunk 1]
```diff
@@ -73,6 +80,10 @@ func GetPrewriteValue(ctx sessionctx.Context, createIfNotExists bool) *binlog.Pr
 
 // WriteBinlog writes a binlog to Pump.
 func (info *BinlogInfo) WriteBinlog(clusterID uint64) error {
+	if atomic.LoadUint32(&errorStopped) != 0 {
+		return terror.ErrCritical.GenByArgs(errors.New("binilog already stopped"))
```
Comment (coocood) [inline sessionctx/binloginfo/binloginfo.go]: If we return an error, how could we still provide service?

Comment (tiancaiamao) [reply sessionctx/binloginfo/binloginfo.go]: The server will not hang now
https://github.com/pingcap/tidb/pull/6503/files#diff-dc417f5322100e2e83a8818c70a25878R481

Comment (coocood) [reply sessionctx/binloginfo/binloginfo.go]: But it can not execute any write statement.

**Comment by tiancaiamao** [comment]
PTAL @coocood

**Comment by coocood** [comment]
@tiancaiamao 
When binlog failed once, we should stop writing binlog anymore.

**Comment by tiancaiamao** [comment]
If we ignore it totally, critical error metrics would not increase, and that's dangerous. @coocood

**Comment by tiancaiamao** [comment]
PTAL @coocood

**Comment by tiancaiamao** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Comment by GregoryIan** [comment]
@LinuxGit @tennix we change binlog configuration item of  tidb-server

**Comment by GregoryIan** [comment]
@tiancaiamao can we provoide a http api to reset error stop flag?

**Comment by iamxy** [comment]
LGTM

**Comment by tiancaiamao** [comment]
I add a http api.
PTAL @coocood @jackysp

**Comment by jackysp** [comment]
LGTM

**Pull request merged**